hamocc/mo_read_fedep.F90

```diff
@@ -76,34 +79,77 @@ subroutine ini_read_fedep(kpie,kpje,omask)
     endif
 
     if (mnproc.eq.1) then
-      write(io_stdo_bgc,*)'Memory allocation for variable dustflx ...'
+      write(io_stdo_bgc,*)'Memory allocation for variable dustflx_tot ...'
       write(io_stdo_bgc,*)'First dimension    : ',kpie
       write(io_stdo_bgc,*)'Second dimension   : ',kpje
       write(io_stdo_bgc,*)'Third dimension    :  12'
     endif
+    allocate (dustflx_tot(kpie,kpje,12),stat=errstat)
+    if(errstat.ne.0) stop 'not enough memory dustflx_tot'
+    dustflx_tot(:,:,:) = 0.0
 
-    allocate (dustflx(kpie,kpje,12),stat=errstat)
-    if(errstat.ne.0) stop 'not enough memory dustflx'
-    dustflx(:,:,:) = 0.0
+    if (mnproc.eq.1) then
+      write(io_stdo_bgc,*)'Memory allocation for variable dustflx_sfe ...'
+      write(io_stdo_bgc,*)'First dimension    : ',kpie
+      write(io_stdo_bgc,*)'Second dimension   : ',kpje
+      write(io_stdo_bgc,*)'Third dimension    :  12'
+    endif
+    allocate (dustflx_sfe(kpie,kpje,12),stat=errstat)
+    if(errstat.ne.0) stop 'not enough memory dustflx_sfe'
+    dustflx_sfe(:,:,:) = 0.0
 
     ! Open netCDF data file
     if (mnproc==1) then
       ncstat = NF90_OPEN(trim(fedepfile),NF90_NOWRITE, ncid)
       if (ncstat /= NF90_NOERR ) then
-        call xchalt('(get_dust: Problem with netCDF1)')
-        stop        '(get_dust: Problem with netCDF1)'
+        call xchalt('(ini_read_fedep: Problem with netCDF1)')
+        stop        '(ini_read_fedep: Problem with netCDF1)'
       end if
     end if
 
     ! Read  data
-    call read_netcdf_var(ncid,'DUST',dustflx(1,1,1),12,0,0)
+    select case(trim(fedep_source))
+
+      case('mahw2006')
+        ! Dust/iron deposition using the Mahowald et al. 2006 model data.
+        ! The variable 'DUST' contains the total dust-flux in kg/m2/month; this
+        ! is converted to kmol/m2/s of bioavailable (soluble) iron using the molar 
+        ! weight of iron (55.85) and assuming that dust contains a fraction 
+        ! 'frac_ironindust' of iron, and that soluble is a fraction 'frac_soliron' of 
+        ! total iron. A tuning factor 'fetune' is applied to the soluble fraction of iron.
+        ! Note the conversion kg/m2/month -> kg/m2/s is assuming 30 days per month.	  
+        call read_netcdf_var(ncid,'DUST',dustflx_tot(1,1,1),12,0,0)
+        dustflx_tot(:,:,:) = dustflx_tot(:,:,:)/30.0/sec_per_day
+        dustflx_sfe(:,:,:) = dustflx_tot(:,:,:)*frac_ironindust*frac_soliron/55.85*fetune
```

Comment:
Can we add the mol weight iron to `mo_chemcon` (i.e. mw_fe=55.85) instead of hard-coding it here?

Reply:
Ok, will do and replace throughout.